### ISSUE TITLE
Link added to topology even when server returned error

### ISSUE THREAD

**Issue by ehlers** [GNS3/gns3-gui#289]
GNS3 V1.3.1, OS Debian Linux 8.0 (Jessie)

See https://community.gns3.com/thread/7971

Scenario to reproduce:
- create a new topology
- add cloud with NIO-TAP
- add switch
- connect cloud to switch
- delete link

Dynamips doesn't close the file descriptor:
```
behlers@dell:~$ sudo lsof -n -a -d0-65536 -p 3456
COMMAND   PID    USER   FD   TYPE DEVICE SIZE/OFF     NODE NAME
dynamips 3456 behlers    0r   CHR    1,3      0t0     1028 /dev/null
dynamips 3456 behlers    1u   REG    8,8        0 19136844 /tmp/tmpoxcof1me
dynamips 3456 behlers    2u   REG    8,8        0 19136844 /tmp/tmpoxcof1me
dynamips 3456 behlers    3w   REG    8,8      832 20447956 /home/behlers/GNS3/projects/test/project-files/dynamips/dynamips_i1_log.txt
dynamips 3456 behlers    4u  IPv4  43163      0t0      TCP *:46118 (LISTEN)
dynamips 3456 behlers    5u  IPv4  43165      0t0      TCP 192.168.1.11:46118->192.168.1.11:35178 (ESTABLISHED)
dynamips 3456 behlers    6u   CHR 10,200      0t0     8918 /dev/net/tun
```
That is something that has to be investigated on the server side.

On the client side the fun starts now:
- connect cloud to switch

Because of the server side problem you get an error message (unable to create TAP NIO) and so no link is shown. But when you save the topology the link is in the .gns3 file.

- now retry this link creation 2 more times

Of cause the link creation fails, but when you save the topology, you now have 3 links from the cloud to the switch. These multiple links will create a lot of problem in this topology.

The same problem can be created by connecting a cloud with NIO TAP to a qemu devices. As qemu (currently) doesn't support NIO TAP, you will get an error. Nevertheless this link is added to the topology. On reload of the topology the user again faces error messages he can't understand.

**Comment by grossmj** [comment]
@ehlers have you tested with the latest dev version? I am saying this because I have fixed am issue with Dynamips which was not properly closing NIOs.

**Comment by ehlers** [comment]
Wow, I just tested with the dev version and the dynamips problem is gone. So the server side is already fixed. That's great.

Nevertheless on the gui side you can still trigger this by connecting a NIO TAP cloud interface to a qemu or VirtualBox VM. The gui gets an error from the server, because it's not implemented. Nevertheless the link is in the topology, what you can see, when you look into the saved file.

**Comment by grossmj** [comment]
Fixed: https://github.com/GNS3/gns3-gui/commit/6792e3d70122f0b0c7a3c87c52134d330d2beeb0 Typo in the commit name, should have been Fixes #289 ;)
